### DISCUSSION TITLE
 ERR_PNPM_PEER_DEP_ISSUES  - or not?

### DISCUSSION THREAD

**Discussion by tomdavidson** [pnpm/pnpm#5366] [Q&A]
The first time I run `pnpm i` after a deb change, I will have   `ERR_PNPM_PEER_DEP_ISSUES `.  When I run `pnpm i` a second time, there are no errors/issues. 

.npmrc has `auto-install-peers=true`

First time trying to adopt pnpm is this how its suppose to work? Im troubleshooting another issue that may more not be related.

**Comment by tomdavidson** [comment]
Now this is a blocker and is consuming quite a bit of time.  Sometimes I think I have it resolved sometimes `pnpm i` yields the error.  It doesnt look like `pnpm i || pnpm i` would be a workaround but Ill try it for a bit. Currently storybook is broken:

```shell
 ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies

packages/ui
├─┬ @storybook/addon-actions 6.5.12
│ └─┬ react-inspector 5.1.1
│   └── ✕ unmet peer react@"^16.8.4 || ^17.0.0": found 18.2.0
├─┬ @storybook/addon-essentials 6.5.12
│ └─┬ @storybook/addon-docs 6.5.12
│   └─┬ @mdx-js/react 1.6.22
│     └── ✕ unmet peer react@"^16.13.1 || ^17.0.0": found 18.2.0
└─┬ @storybook/react 6.5.12
  └─┬ react-element-to-jsx-string 14.3.4
    ├── ✕ unmet peer react@"^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1": found 18.2.0
    └── ✕ unmet peer react-dom@"^0.14.8 || ^15.0.1 || ^16.0.0 || ^17.0.1": found 18.2.0
```

I have a `.pnpfile.cjs` configured with the following readPackage hook:

```js
function readPackage(pkg, context) {
  return pkg.name === 'ui'
    ? {
        ...pkg,
        dependencies: {
          ...pkg.dependencies,
          react: '^17.0.1',
          'react-dom': '^17.0.1',
        },
      }
    : pkg;
}

module.exports = {
  hooks: {
    readPackage,
  },
};
```

**Reply by zkochan** [reply]
Yes, currently it prints the peer dependency related errors only when creatin a lockfile.

**Reply by tomdavidson** [reply]
Ah gesh, confirmation bias led me to assume it was user error rather than an issue with pnpm. Will update and report back/close. Thanks.

**Comment by tomdavidson** [comment]
Hello @zkochan so the error is real, but magically goes way if I run `pnpm i` a second time? and shouldn't auto install peers pull in the unmet peers?

**Reply by zkochan** [reply]
yes, it should
but there were some edge cases that I recently fixed. Try with v7.12.1

**Reply by tomdavidson** [reply]
@zkochan shucks, still getting the same errors, with 7.12.1 and 7.12.2
